### PR TITLE
Add docs for using specs2 for testing.

### PR DESCRIPTION
**Pull request by pmellati**
Instructions for specs2 as a custom testing framework.

### CODE REVIEW

File: docs/pages/2 - Configuring Mill.md [hunk 1]
```diff
@@ -7,11 +7,11 @@ import mill._, scalalib._
 
 object foo extends ScalaModule {
   def scalaVersion = "2.12.4"
-
```
Comment (pmellati) [inline docs/pages/2 - Configuring Mill.md]: A few trailing whitespaces are automatically removed by my editor.

File: docs/pages/2 - Configuring Mill.md [hunk 2]
```diff
@@ -182,21 +182,46 @@ configuration options apply.
...
+
+object foo extends ScalaModule {
+  def scalaVersion = "2.12.4"
+
+  object test extends Tests {
+    def ivyDeps = Agg(ivy"org.specs2::specs2-core:4.6.0")
+    def testFrameworks = Seq("org.specs2.runner.Specs2Framework")
+    def scalacOptions = foo.scalacOptions() ++ Seq(
+      "-Yrangepos"
+    )
```
Comment (lefou) [inline docs/pages/2 - Configuring Mill.md]: Why is that? Is that needed to run specs2? If so, could you please add a comment to the example?

Comment (pmellati) [reply docs/pages/2 - Configuring Mill.md]: Yes, needed by specs2: https://etorreborre.github.io/specs2/website/SPECS2-4.6.0/quickstart.html

Comment added in code to clarify.

**Comment by lefou** [comment]
Thanks!

**Pull request merged**